DNN Platform/Dnn.AuthServices.Jwt/Components/Common/Controllers/JwtController.cs

```diff
@@ -0,0 +1,489 @@
+            }
+            ptoken.TokenExpiry = expiry;
+
+            var portalSettings = PortalController.Instance.GetCurrentPortalSettings();
+            var secret = ObtainSecret(ptoken.TokenId, portalSettings.GUID, userInfo.Membership.LastPasswordChangeDate);
+            var jwt = CreateJwtToken(secret, portalSettings.PortalAlias.HTTPAlias, ptoken, userInfo.Roles);
+            var accessToken = jwt.RawData;
+            
+            // save hash values in DB so no one with access can create JWT header from existing data
+            ptoken.TokenHash = GetHashedStr(accessToken);
+            DataProvider.UpdateToken(ptoken);
+
+            return new LoginResultData
+            {
+                DisplayName = userInfo.DisplayName,
+                AccessToken = accessToken,
+                RenewalToken = renewalToken
+            };
+        }
+
+        #endregion
+
+        #region private methods
+
+        private static string NewGuid => Guid.NewGuid().ToString("N");
+
+        private static LoginResultData EmptyWithError(string error)
+        {
+            return new LoginResultData { Error = error };
+        }
+
+        private static JwtSecurityToken CreateJwtToken(byte[] symmetricKey, string issuer, PersistedToken ptoken, IEnumerable<string> roles)
+        {
+            //var key = Convert.FromBase64String(symmetricKey);
+            var credentials = new SigningCredentials(
+                new InMemorySymmetricSecurityKey(symmetricKey),
+                "http://www.w3.org/2001/04/xmldsig-more#hmac-sha256",
+                "http://www.w3.org/2001/04/xmlenc#sha256");
+
+            var claimsIdentity = new ClaimsIdentity();
+            claimsIdentity.AddClaim(new Claim(SessionClaimType, ptoken.TokenId));
+            claimsIdentity.AddClaims(roles.Select(r => new Claim(ClaimTypes.Role, r)));
+
+            var notBefore = DateTime.UtcNow.AddMinutes(-ClockSkew);
+            var notAfter = ptoken.TokenExpiry;
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var token = tokenHandler.CreateToken(issuer, null, claimsIdentity, notBefore, notAfter, credentials);
+            return token;
+        }
+
+        /// <summary>
+        /// Checks for Authorization header and validates it is JWT scheme. If successful, it returns the token string.
+        /// </summary>
+        /// <param name="authHdr">The request auhorization header.</param>
+        /// <returns>The JWT passed in the request; otherwise, it returns null.</returns>
+        private string ValidateAuthHeader(AuthenticationHeaderValue authHdr)
+        {
+            if (authHdr == null)
+            {
+                //if (Logger.IsTraceEnabled) Logger.Trace("Authorization header not present in the request"); // too verbose; shows in all web requests
+                return null;
+            }
+
+            if (!string.Equals(authHdr.Scheme, AuthScheme, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
why is this a culture-specific comparison?  I would expect `OrdinalIgnoreCase` to be used here…